source/common/router/upstream_request.cc

```diff
@@ -213,16 +213,18 @@ void UpstreamRequest::decodeTrailers(Http::ResponseTrailerMapPtr&& trailers) {
 void UpstreamRequest::dumpState(std::ostream& os, int indent_level) const {
   const char* spaces = spacesForLevel(indent_level);
   os << spaces << "UpstreamRequest " << this << "\n";
-  const auto addressProvider = connection().connectionInfoProviderSharedPtr();
+  if (connection()) {
```

Comment:
again this appears to be a semantic change; was this previously a potential crasher?